### PR TITLE
NocoDB Node updated for v0.90.0 - N8N-3411

### PR DESCRIPTION
**Pull request by mertmit**
- Added an option to use new API for NocoDB. (Old API will be deprecated on 30 Jun 2022)

### CODE REVIEW

**Comment by o1lab** [comment]
With NocoDB version >0.90 the API docs have changed. @janober - if you could pls see the merge through it will help the community. @mertmit can help follow up with any comments to the PR. Thank you

**Comment by ed-parsadanyan** [comment]
Hello. As mentioned in [#3170](https://github.com/n8n-io/n8n/issues/3170) the issue is still here.
From the commit code I would expect new options in the NocoDB Node (see screenshot), but in the current Docker version of n8n (174.0) I don't see any changes in the interface.

image

**Comment by ed-parsadanyan** [comment]
> > but in the current Docker version of n8n (0.174.0)
> 
> This PR is yet to be merged - did you expect to still see this in 0.174.0 ?

Oh, I see now, it wasn't merged yet. Hope it will be completed for the next n8n release.

**Comment by Joffcom** [comment]
Hey @o1lab & @ed-parsadanyan,

This one has not been reviewed yet it has been on my list for a while but we have been chatting internally about how to deal with API changes like this in a way that can work for existing and future nodes.

I am going to be taking a look today to see how this works out with our new approach.

**Comment by o1lab** [comment]
@Joffcom : Awesome  - thank you.

**Comment by fishnux** [comment]
Any progress on this review for merge, @Joffcom? Thank you!

**Comment by Joffcom** [comment]
Hey @fishnux,

I was off last week but the good news is we have worked out how to handle API versioning in a way that should scale for all nodes in the future. I will have a few tweaks to make to this which are at the top of my list of things to do so we should be have this mostly wrapped up this week.

File: packages/nodes-base/nodes/NocoDB/GenericFunctions.ts [hunk 1]
```diff
@@ -35,14 +35,28 @@ interface IAttachment {
...
 export async function apiRequest(this: IHookFunctions | IExecuteFunctions | ILoadOptionsFunctions | IPollFunctions, method: string, endpoint: string, body: object, query?: IDataObject, uri?: string, option: IDataObject = {}): Promise<any> { // tslint:disable-line:no-any
-	const credentials = await this.getCredentials('nocoDb');
+	const authenticationMethod = this.getNodeParameter('authentication', 0);
+	let credentials;
+
+	if (authenticationMethod === 'nocoDbApiToken') {
+		credentials = await this.getCredentials('nocoDbApiToken');
+	} else if (authenticationMethod === 'nocoDb') {
+		credentials = await this.getCredentials('nocoDb');
+	}
```
Comment (janober) [inline packages/nodes-base/nodes/NocoDB/GenericFunctions.ts:45]: Can we not simply do the following here?
`credentials = await this.getCredentials(authenticationMethod);`

File: packages/nodes-base/nodes/NocoDB/GenericFunctions.ts [hunk 2]
```diff
@@ -55,8 +69,10 @@ export async function apiRequest(this: IHookFunctions | IExecuteFunctions | ILoa
 		delete options.body;
 	}
 
+	const credentialType = authenticationMethod === 'nocoDb' ? 'nocoDb' : 'nocoDbApiToken';
+
```
Comment (janober) [inline packages/nodes-base/nodes/NocoDB/GenericFunctions.ts:73]: Is this necessary? Is the value of `credentialType` not simply again the value of `authenticationMethod` afterward?

File: packages/nodes-base/nodes/NocoDB/GenericFunctions.ts (hunk 1 above)
Comment (Joffcom) [reply packages/nodes-base/nodes/NocoDB/GenericFunctions.ts]: Changed 👍🏻

File: packages/nodes-base/nodes/NocoDB/GenericFunctions.ts (hunk 2 above)
Comment (Joffcom) [reply packages/nodes-base/nodes/NocoDB/GenericFunctions.ts]: It is, Good catch. Have now changed it.

**Comment by janober** [comment]
Thanks a lot. Got merged.

**Comment by janober** [comment]
Got released with `n8n@0.189.0`

**Pull request merged**